Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -0,0 +1,877 @@
+    $reqMDCTier = "Standard";
+    $reqProviderName = "Microsoft.Security"
+    $isProviderRegister = $true
+    $previousProviderRegistrationState = $false
+
+    
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 2 of 4]: Checking [$($reqMDCTier)] pricing tier for required resource types..."
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Checking IsProviderRegister with 'Microsoft.Security' provider
+    $registeredProvider =  Get-AzResourceProvider -ProviderNamespace $reqProviderName | Where-Object { $_.RegistrationState -eq "Registered" }
+
+    if($null -eq $registeredProvider)
+    {
+        # capture provider registration state
+        $isProviderRegister = $false
+        Write-Host "Found [$($reqProviderName)] provider is not registered."  -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host "$reqProviderName registering [It takes 2-3 min to get registered]..." -ForegroundColor $([Constants]::MessageType.Info)
+        # Registering provider with required provider name, it will take 1-2 min for registration
+        try 
+        {
+            $provider = Register-AzResourceProvider -ProviderNamespace $reqProviderName
+            Write-Host "[$reqProviderName] registering..." -ForegroundColor $([Constants]::MessageType.Info)
+            while((((Get-AzResourceProvider -ProviderNamespace $reqProviderName).RegistrationState -ne "Registered") | Measure-Object).Count -gt 0)
+            {
+                # Checking threshold time limit to avoid getting into infinite loop
+                if($thresholdTimeLimit -ge 300)
+                {
+                    Write-Host "Error occurred while registering [$($reqProviderName)] provider. It is taking more time than expected, Aborting process..." -ForegroundColor $([Constants]::MessageType.Error)
+                    throw [System.ArgumentException] ($_)
+                }
+                
+                Start-Sleep -Seconds 30
+               
+                # Incrementing threshold time limit by 30 sec in every iteration
+                $thresholdTimeLimit = $thresholdTimeLimit + 30
+            }
+            $isProviderRegister = $true
+        }
+        catch 
+        {
+            Write-Host "Error occurred while registering $reqProviderName provider. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+            return
+        }
+        Write-Host "$reqProviderName provider successfully registered." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else {
+        $previousProviderRegistrationState = $true
+    }
+
+    $nonCompliantMDCTierResourcetype = @()
+    $nonComplaintResourceType = Get-AzSecurityPricing | Where-Object { $_.PricingTier -ne $reqMDCTier -and $reqMDCTierResourceTypes.Contains($_.Name) } | select "Name", "PricingTier", "Id"
```

Comment:
we should add check for subplan as well?

Reply:
Done